src/app/components/shared/model-cards/card/card.component.ts

```diff
@@ -90,43 +38,65 @@ import { WithLoadingPipe } from "../../../../pipes/with-loading/with-loading.pip
     AsyncPipe,
     WithLoadingPipe,
   ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CardComponent implements OnInit {
+export class CardComponent {
   private readonly recordingApi = inject(AudioRecordingsService);
   private readonly session = inject(BawSessionService);
   private readonly licenseService = inject(LicensesService);
+  private readonly projectService = inject(ProjectsService);
 
-  @Input() public model: Project | Region;
+  public readonly model = input<Project | Region>();
 
-  public hasNoAudio$: Observable<boolean>;
-  protected isOwner: boolean;
-  protected licenseText: string | undefined = undefined;
+  protected readonly licenseResource = resource({
+    params: () => ({ cardModel: this.model() }),
+    loader: ({ params }): Promise<string | null> => {
+      const cardModel = params.cardModel;
+      if (cardModel instanceof Project) {
+        return Promise.resolve(cardModel.license);
+      }
 
-  public ngOnInit(): void {
-    this.isOwner = this.model.creatorId === this.session.loggedInUser?.id;
-    this.hasNoAudio$ = this.getRecordings().pipe(
-      map((recordings): boolean => recordings.length === 0),
-    );
-    this.updateLicense();
-  }
+      // Because regions can only have a maximum of one project, it is safe to
+      // take the first result.
+      // However, because regions can become orphaned, I need a nullish check
+      // before accessing the license property so that if no project is found,
+      // we do not throw an error.
+      //
+      // I also use nullish coalescing to return null if no license is found
+      // so that we don't mix undefined and null return types when there are no
+      // projects associated with the region.
+      const projectLicense = this.projectService.getProjectFor(cardModel).pipe(
+        map((project) => project[0]?.license ?? null),
+      );
+
+      return firstValueFrom(projectLicense);
+    },
+  });
+
+  protected readonly licenseTextResource = resource({
+    params: () => ({ license: this.licenseResource.value() }),
+    loader: async ({ params }) => {
+      return await this.licenseService.licenseText(params.license);
+    },
+  });
```

Comment:
seems like this could be combined with licenseResource.

do async await. No reason to have more signals if we don't need to branch off of them

Reply:
<img width="686" height="525" alt="image" src="https://github.com/user-attachments/assets/7b450f3a-1304-406b-8cf0-8b3becb56949" />